File: DAY1/Assignment/Quesn2/Quesn2.cpp
//Print Linked List
#include <iostream>

using namespace std;

class node{
    public:
            int data;
            node* next;

            node(int val){
                data = val;
                next = NULL;
            }
};

void insertAttail(node* &head, int val){
    node* n = new node(val);

    if(head == NULL){
        head = n;
        return;
    }
    //traverse
    node* temp = head;

    while(temp->next != NULL){
        temp = temp->next;
    }
    temp->next = n;
}

void printll(node* head){
    node* temp = head;

    while(temp != NULL){
        cout<<temp->data<<"->";
        temp = temp->next;
    }
    cout<<"NULL"<<endl;
}

int main(){
    node* head = NULL;
    insertAttail(head,2);
    insertAttail(head,3);
    insertAttail(head,5);
    printll(head);
    return 0;
}

File: WEEK3/DAY1/Assignment/Quesn2/Quesn2.java
//Print Linked List
public class LinkedList{
    Node head;
    static class Node{
        int data;
        Node next;

        Node(int val){
            data = val;
            next = null;
        }
    }

    public static LinkedList insertAttail(LinkedList list, int val){
        Node n = new Node(val);
        n.next = null;

        if(list.head == null){
            list.head = n;
        }
        else{
            Node temp = list.head;

            while(temp.next != null){
                temp = temp.next;
            }

            temp.next = n;
        }
        return list;
    }
    
    public static void printll(LinkedList list){
        Node temp = list.head;

        while(temp != null){
            System.out.print(temp.data + " ");

            temp = temp.next;
        }
    }

    public static void main(String[] args) {
        LinkedList list = new LinkedList();

        list = insertAttail(list, 2);
        list = insertAttail(list, 3);
        list = insertAttail(list, 5);
        printll(list);
    }

}
